packages/legacy/core/App/screens/Splash.tsx

```diff
@@ -230,6 +230,32 @@ const Splash: React.FC = () => {
           return
         }
 
+        if (agent) {
+          logger.info('Agent already initialized, restarting...')
+
+          try {
+            await agent.wallet.open({
+              id: walletSecret.id,
+              key: walletSecret.key,
+            })
+          } catch (error: unknown) {
+            logger.error('Error opening existing wallet', error as Error)
```

Comment:
Should we not catch here and let it become a BifoldError that gets emitted like the rest of the error handling in this func? Otherwise it may silently fail, no?